src/contracts/middleware/IndexRegistry.sol

```diff
@@ -235,19 +236,18 @@ contract IndexRegistry is IIndexRegistry {
 
     /// @notice Returns the index of the `operatorId` at the given `blockNumber` for the given `quorumNumber`, or max uint32 if the operator is not active in the quorum
     function _getIndexOfOperatorForQuorumAtBlockNumber(bytes32 operatorId, uint8 quorumNumber, uint32 blockNumber) internal view returns(uint32) {
-        OperatorIndexUpdate memory operatorIndexUpdate;
-        // set to max uint32 value to indicate that the operator is not part of the quorum at all, until this is updated in the loop
-        operatorIndexUpdate.index = type(uint32).max; 
+        uint256 operatorIndexHistoryLength = _operatorIdToIndexHistory[operatorId][quorumNumber].length;
         // loop forward through index history to find the index of the operator at the given block number
         // this is less efficient than looping backwards, but is simpler logic and only called in view functions that aren't mined onchain
         for (uint i = 0; i < _operatorIdToIndexHistory[operatorId][quorumNumber].length; i++) {
-            operatorIndexUpdate = _operatorIdToIndexHistory[operatorId][quorumNumber][i];
+            uint256 listIndex = (operatorIndexHistoryLength - 1) - i;
+            OperatorIndexUpdate memory operatorIndexUpdate = _operatorIdToIndexHistory[operatorId][quorumNumber][listIndex];
             if (operatorIndexUpdate.fromBlockNumber <= blockNumber) {
                 return operatorIndexUpdate.index;
             }
         }
 
         // the operator is still active or not in the quorum, so we return the latest index or the default max uint32
-        return operatorIndexUpdate.index;
+        return type(uint32).max;
```

Comment:
won't we now hit this only if the operator wasn't ever active before the blockNumber? something seems wrong here

Reply:
We will only hit this, if, after looping backwards through all of the updates, none of them have a fromBlockNumber before the given blockNumber

Reply:
OK. agreed the behavior is fine now, but you should update the comment. I believe this should get hit:
1) if the `blockNumber` is from before the operator registered
2) if the operator never registered (technically overlaps with above, but worth calling out separately)
Let me know if you disagree.

I'd also clarify this in the top-level comment for the function